api/app/db/crud/weather_models.py

```diff
@@ -244,75 +244,62 @@ def get_station_model_predictions(
 def get_latest_station_model_prediction_per_day(session: Session,
                                                 station_codes: List[int],
                                                 model: str,
-                                                start_date: datetime.datetime,
-                                                end_date: datetime.datetime):
+                                                day_start: datetime.datetime,
+                                                day_end: datetime.datetime):
+    """
+    All predictions for a given day for each station, ordered by update_timestamp
+    """
     result = session.execute(f"""
```

Comment:
could you please add another code comment with a more detailed description of what this query is doing? It's so complex it's difficult to follow, esp. if we later have to debug it.

Reply:
or a description somewhere. Doesn't have to be a code comment - wiki or readme would be great too

Reply:
I updated the comment in: https://github.com/bcgov/wps/pull/2675/commits/92c2dd747c3fdbbb739e66ce31b1cd68353ca769

However the complexity of the query is inherent to the existing table design; if we want to simplify it we'd need to redesign it through denormalization or some other structure.

Reply:
I'm not worried about the complexity of the query at this time, just want any dev to be able to understand what the query is doing reasonably quickly so that any one of us can return to it later and simplify it if possible or at least be able to debug it if an issue arises.